File: code_samples/swift/data_structures_hash_table.swift
import Foundation

/// Data Structures: Hash Table
/// AI/ML Training Sample

class HashTable {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = HashTable()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")

File: code_samples/rust/data_structures_hash_table.rs
//! Data Structures: Hash Table
//! AI/ML Training Sample

pub struct HashTable {
    data: String,
}

impl HashTable {
    pub fn new() -> Self {
        Self {
            data: String::new(),
        }
    }
    
    pub fn process(&mut self, input: &str) {
        self.data = input.to_string();
    }
    
    pub fn get_data(&self) -> &str {
        &self.data
    }
    
    pub fn validate(&self) -> bool {
        !self.data.is_empty()
    }
}

fn main() {
    let mut instance = HashTable::new();
    instance.process("example");
    println!("Data: {}", instance.get_data());
    println!("Valid: {}", instance.validate());
}
